contracts/VotingMachines/GenesisProtocol.sol

```diff
@@ -889,4 +901,21 @@ contract GenesisProtocol is IntVoteInterface,UniversalScheme {
         ProposalState pState = proposals[_proposalId].state;
         return ((pState == ProposalState.PreBoosted)||(pState == ProposalState.Boosted)||(pState == ProposalState.QuietEndingPeriod));
     }
+
+    function binarySearch(uint[] _data,uint _start,uint _end,uint _value) private view returns(uint) {
```

Comment:
Need to check of `_data` gets pushed to the stack each time. If it is, there's a risk of stack overflow.